docs/datatype/image/text2image.md

```diff
@@ -1,3 +1,484 @@
+
+```python
+import glob
+from jina import Document
+
+def input_docs(data_path):
+    for fn in glob.glob(os.path.join(data_path, '*')):
+        doc = Document(uri=fn, tags={'filename': fn})
+        doc.convert_image_uri_to_blob()
+        yield doc 
+```
+
+The final step in this section is to send the input documents to the index Flow. Note that indexing can take a while:
+
+```python
+    with flow_index:
+        flow_index.post(on='/index',inputs=input_docs("/content/images"), request_size=1)
+```
+
+
+```text
+Flow@3084[I]:🎉 Flow is ready to use!
+🔗 Protocol: 		GRPC
+🏠 Local access:	0.0.0.0:33367
+🔒 Private network:	172.28.0.2:33367
+🌐 Public address:	34.125.186.176:33367
+```
+
+### Searching
+Now, let's build the search Flow and use it in order to find sample query images.
+
+Our Flow contains the following executors:
+
+1. CLIPTextEncoder: We specify the device.
+2. SimpleIndexer : We need to specify the workspace parameter.
+
+```python
+flow_search = Flow().add(uses=CLIPTextEncoder, name="encoder", uses_with={"device":device}) \
+        .add(uses=SimpleIndexer,name="indexer",workspace="workspace")
+flow_search.plot()
+```
+
+Query Flow:
+
+```{figure} query_flow_text2image.svg
+:align: center
+```
+
+We create a helper function to be able to plot our images 
+
+```python 
+import matplotlib.pyplot as plt
+
+def show_docs(docs):
+  for doc in docs:
+      plt.imshow(doc.blob)
+      plt.show()
+```
+and one last function to show us the TOP 3 matches to our text query 
+
+```python 
+def plot_search_results(resp: Request):
+    for doc in resp.docs:
+        print(f'Query text: {doc.text}')
+        print(f'Matches:')
+        print('-'*10)
+        show_docs(doc.matches[:3])
+```        
+
+Now we put some texts queries which we transform into Documents and here's the results: 
+
+```python
+with flow_search:
+    resp = flow_search.post(on='/search',inputs=DocumentArray([
+                    Document(text='dog'),
+                    Document(text='cat'),
+                    Document(text='kids on their bikes'),
+                ]),on_done=plot_search_results)
+```
+
+Sample results: 
+
+```text
+Query: Dog
+Results:
+```
+```{figure} dog1.png
+:align: center
+```
+```{figure} dog2.png
+:align: center
+```
+```{figure} dog3.png
+:align: center
+```
+
+```text
+Query: Cat
+Results:
+```
+
+```{figure} cat1.png
+:align: center
+```
+```{figure} cat2.png
+:align: center
+```
+```{figure} cat3.png
+:align: center
+```
+
+```text
+Query: Kids riding bikes
+Results:
+```
+```{figure} bike1.png
+:align: center
+```
+```{figure} bike2.png
+:align: center
+```
+```{figure} bike3.png
+:align: center
+```
+
+Congratulations you have built a text2image search engine, you can check the full source code [here](https://colab.research.google.com/drive/1CkT1udBrMfefYo0XJO45kIpW43odLLDm?usp=sharing) and experiment with your own text queries.
```

Comment:
- Congratulations! You have built a text-to-image search engine. You can check the full ...